src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchFailuresSpec.groovy

```diff
@@ -71,105 +75,102 @@ class SwitchFailuresSpec extends HealthCheckSpecification {
         given: "A flow"
         def swPair = switchPairs.all().nonNeighbouring().withAtLeastNPaths(2).random()
-        def flow = flowHelperV2.addFlow(flowHelperV2.randomFlow(swPair))
+        def flow = flowFactory.getRandom(swPair)
 
         when: "Current path breaks and reroute starts"
         switchHelper.shapeSwitchesTraffic([swPair.dst], new TrafficControlData(3000))
-        def islToBreak = pathHelper.getInvolvedIsls(northbound.getFlowPath(flow.flowId)).first()
+        def islToBreak = flow.retrieveAllEntityPaths().flowPath.getInvolvedIsls().first()
         antiflap.portDown(islToBreak.srcSwitch.dpId, islToBreak.srcPort)
 
         and: "Switch reconnects in the middle of reroute"
         Wrappers.wait(WAIT_OFFSET, 0) {
-            def reroute = flowHelper.getEarliestHistoryEntryByAction(flow.flowId, REROUTE_ACTION)
+            def reroute = flow.retrieveFlowHistory().getEntriesByType(FlowActionType.REROUTE).first()
             assert reroute.payload.last().action == "Started validation of installed non ingress rules"
         }
         lockKeeper.reviveSwitch(swPair.src, lockKeeper.knockoutSwitch(swPair.src, RW))
 
         then: "Flow reroute is successful"
         Wrappers.wait(PATH_INSTALLATION_TIME * 2) { //double timeout since rerouted is slowed by delay
-            assert northboundV2.getFlowStatus(flow.flowId).status == FlowState.UP
-            assert flowHelper.getLatestHistoryEntry(flow.flowId).payload.last().action == REROUTE_SUCCESS
+            assert flow.retrieveFlowStatus().status == FlowState.UP
+            assert flow.retrieveFlowHistory().getEntriesByType(FlowActionType.REROUTE).last().payload.last().action == FlowActionType.REROUTE.payloadLastAction
         }
 
         and: "Blinking switch has no rule anomalies"
         !switchHelper.validateAndCollectFoundDiscrepancies(swPair.src.dpId).isPresent()
 
         and: "Flow validation is OK"
-        northbound.validateFlow(flow.flowId).each { assert it.asExpected }
+        flow.validateAndCollectDiscrepancies().isEmpty()
     }
 
     def "System can handle situation when switch reconnects while flow is being created"() {
         when: "Start creating a flow between switches and lose connection to src before rules are set"
         def (Switch srcSwitch, Switch dstSwitch) = topology.activeSwitches
-        def flow = flowHelperV2.randomFlow(srcSwitch, dstSwitch)
-        flowHelperV2.attemptToAddFlow(flow)
+        def flow = flowFactory.getBuilder(srcSwitch, dstSwitch).build().sendCreateRequest()
         sleep(50)
-        def blockData = lockKeeper.knockoutSwitch(srcSwitch, RW)
-        cleanupManager.addAction(REVIVE_SWITCH, {switchHelper.reviveSwitch(srcSwitch, blockData)})
+        def blockData = switchHelper.knockoutSwitch(srcSwitch, RW)
 
         then: "Flow eventually goes DOWN"
         Wrappers.wait(WAIT_OFFSET) {
             assert northbound.getSwitch(srcSwitch.dpId).state == SwitchChangeType.DEACTIVATED
         }
```

Comment:
This check is included in `switchHelper.knockoutSwitch`

Reply:
Done: removed this check